clippy_lints/src/methods/mod.rs

```diff
@@ -3563,11 +3566,47 @@ declare_clippy_lint! {
     "calls to `.take()` or `.skip()` that are out of bounds"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Looks for calls to `Path::ends_with` calls where the argument looks like a file extension.
+    ///
+    /// By default, Clippy has a short list of known filenames that start with a dot
+    /// but aren't necessarily file extensions (e.g. the `.git` folder), which are allowed by default.
+    /// File extensions are often two to three characters long, so this only lints in those cases
+    /// in an attempt to avoid false positives.
+    /// The `allowed-file-extensions` configuration can be used to allow additional
+    /// file extensions that Clippy should not lint.
+    ///
+    /// ### Why is this bad?
+    /// This doesn't actually compare file extensions. Rather, `ends_with` compares the given argument
+    /// to the last **component** of the path and checks if it matches exactly.
+    ///
+    /// ### Example
```

Comment:
Probably worth sticking a known issues section in that mentions it won't pick up longer extension names, with that and a squash it looks good to go

Reply:
This is already mentioned in "What it does", but I guess that can be moved to its own "Known issues" section